error.go

```diff
@@ -1,45 +1,22 @@
 package uuid
 
-import (
-	"fmt"
+type Error string
+
+const (
+	ErrInvalidFormat           = Error("uuid: invalid UUID format")
+	ErrIncorrectFormatInString = Error("uuid: incorrect UUID format in string")
+	ErrIncorrectLength         = Error("uuid: incorrect UUID length")
+	ErrIncorrectByteLength     = Error("uuid: UUID must be exactly 16 bytes long")
+	ErrNoHwAddressFound        = Error("uuid: no HW address found")
+	ErrTypeConvertError        = Error("uuid: cannot convert")
+	ErrInvalidVersion          = Error("uuid:")
 )
 
-// ErrUUID is the base error type for errors generated by this package
-type ErrUUID struct{}
-
-func (e *ErrUUID) Error() string {
-	return "uuid"
-}
-
-// ErrUUIDInvalidFormat is a specialized error that is returned when there is
-// a problem parsing UUID data
-type ErrUUIDInvalidFormat struct {
-	err error
-}
-
-func (e *ErrUUIDInvalidFormat) Error() string {
-	return e.err.Error()
-}
-
-func (e *ErrUUIDInvalidFormat) Unwrap() error {
-	return e.err
+func (e Error) Error() string {
+	return string(e)
 }
 
-func (e *ErrUUIDInvalidFormat) Is(target error) bool {
-	_, ok := target.(*ErrUUIDInvalidFormat)
+func (e Error) Is(target error) bool {
```

Comment:
Is this explicit `Is()` method necessary?  I thought `errors.Is()` already includes this same logic and you only need the `Is()` method is you're doing something non-standard.

Reply:
I believe it is needed, the code doesn't work otherwise. But I can do some more research perhaps.

Reply:
You definitely don't need to provide `Is()` yourself.  [This playground](https://go.dev/play/p/jaCtGzLYIrS) shows that `errors.Is()` and `errors.As()` both work for plain and wrapped errors without needing explicit `Is()` or `As()` methods.

Reply:
With the code you provided in the playground, no it's not needed. But with the code in my tests it's needed.
It is needed in order to check if the error returned is of type uuid.Error.
Like this: `if errors.Is(err, &fooError)`
Otherwise you can only user errors.As() and it is not always that you want to get the actual error object. Especially in this case since the error object doesn't contain any additional information. It is more interesting to know if the error is a uuid.Error type (doesn't matter if it is a ErrIncorrectFormatInString or ErrIncorrectLength or whatever...)
If you read the documentation of `errors.Is()` they give an example of this specific usage.

Reply:
I see what's happening.  `errors.Is()` checks whether or not `err` is (or wraps) a *specific* sentinel error, not whether it is of a given type. Use `errors.As()` to check if an error is of a certain type.
```
// was the error specifically ErrInvalidFormat?
if errors.Is(err, uuid.ErrInvalidFormat) {
   ...
}
// is the concrete type of err uuid.Error?
// - note that this is uuid.Error, *not* *uuid.Error because the declared constants aren't pointers
var ue uuid.Error
if errors.As(err, &ue) {
   ...
}
```

Reply:
Ok, so you don't think the `Is()` function should be provided, but the user should use `As()` instead?

Reply:
Ok I see. I guess that's a valid point of view, but it is a bit different from how the standard lib does it. Here is an example:
```
func (e Errno) Is(target error) bool {
	switch target {
	case oserror.ErrPermission:
		return e == EACCES || e == EPERM
	case oserror.ErrExist:
		return e == EEXIST || e == ENOTEMPTY
	case oserror.ErrNotExist:
		return e == ENOENT
	case errorspkg.ErrUnsupported:
		return e == ENOSYS || e == ENOTSUP || e == EOPNOTSUPP
	}
	return false
}
```
Since I don't think I can convince you, and since it's easy to implement again in the future I guess I should remove it for now.

Reply:
Correct. That `Is()` method is comparing `target` for equality against a specific, finite set of sentinel errors.  Nothing in that logic involves the concrete type of `target`.